File: arrays_string/isunique/sol1.py

def isunique(a):
 marker = 0
 bl = ord('a')
 for item in a:
  b = 1 << (ord(item) - bl)
  if marker & b > 0:
   return False
  else:
   marker |= b
 return True
 
 
if __name__ == "__main__":
 test_str = ['aaaaa', 'abcdea', 'abcdefg']
 test_sol = [False, False, True]
 
 for i in xrange(len(test_str)):
  sol = isunique(test_str[i])
  if sol == test_sol[i]:
   print test_str[i] + " passes " + str(sol)
  else:
   print test_str[i] + " no pass " + str(sol)

File: arrays_string/isunique/sol1.cpp
#include <iostream>
#include <string>
#include <vector>

bool isunqiue(const std::string& str) {
  int a_ord = 'a';
  int res = 0;
  for (auto& c : str) {
    int b = c;
    b = 1 << b;
    if (res & b) {
      return false;
    }
    res |= b;
  }
  return true;
}


int main() {
  std::vector<std::string> test_str{"aaaaa", "abcdea", "abcdefg"};
  for(auto& s : test_str) {
    std::cout << s << " ";
    if (isunqiue(s)) {
      std::cout << "unique";
    } else {
      std::cout << "not unique";
    }
    std::cout << std::endl;
  }
  return 0;
}